applications/HDF5Application/custom_utilities/vertex_utilities.cpp

```diff
@@ -0,0 +1,55 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 license: HDF5Application/license.txt
+//
+//  Main author:     Máté Kelemen
+//
+
+// Internal includes
+#include "vertex_utilities.h"
+
+
+namespace Kratos
+{
+namespace HDF5
+{
+
+
+BruteForcePointLocatorAdaptor::BruteForcePointLocatorAdaptor(ModelPart& rModelPart,
+                                                             const Globals::Configuration configuration,
+                                                             const double tolerance)
+    : mLocator(rModelPart),
+      mrModelPart(rModelPart),
+      mConfiguration(configuration),
+      mTolerance(tolerance)
+{
+}
+
+
+const Element::WeakPointer BruteForcePointLocatorAdaptor::FindElement(const Point& rPoint) const
+{
+    KRATOS_TRY
+
+    Kratos::Vector shape_function_values;
+
+    const int element_id =  mLocator.FindElement(
```

Comment:
FYI this is super expensive and should be used as few times as possible

Reply:
This is only called when a ```Vertex``` is constructed, and that only happens during the initialization of ```PointSetOutput``` so I think it's as little used as possible. I made an interface (```PointLocatorAdaptor```) if someone wants to implement and use a different point locator. I know there's ```BinBasedPointLocator``` but I haven't used it yet.